src/repr/sdd.rs

```diff
@@ -211,54 +211,42 @@ impl SddPtr {
         vtree: &VTreeManager,
         map: &WmcParams<FiniteField<P>>,
     ) -> FiniteField<P> {
-        if self.is_true() {
-            return FiniteField::new(1);
-        }
-        if self.is_false() {
-            return FiniteField::new(0);
-        }
-        if self.is_or() {
-            let r = self.node_ref_mut();
-            if r.semantic_hash.is_some() {
-                let v = FiniteField::new(r.semantic_hash.unwrap());
-                return if self.is_neg() { v.negate() } else { v };
+        match self {
+            PtrTrue => FiniteField::new(1),
+            PtrFalse => FiniteField::new(0),
+            Var(v, b) => {
+                let (l_w, h_w) = map.get_var_weight(*v);
+                return if *b { *h_w } else { *l_w };
             }
-            // no cached value, compute it
-            let h = self.node_iter().fold(FiniteField::new(0), |acc, i| {
-                acc + i.prime().cached_semantic_hash(vtree, map)
-                    * i.sub().cached_semantic_hash(vtree, map)
-            });
-            self.node_ref_mut().semantic_hash = Some(h.value());
-            if self.is_neg() {
-                return h.negate();
-            } else {
-                return h;
-            };
-        }
-        if self.is_var() {
-            let v = self.get_var();
-            let (h_w, l_w) = map.get_var_weight(v.get_label());
```

Comment:
In every other place in the code, this is `l_w`, `h_w`. Flipped it - but that didn't resolve the problem.